test/e2e/compact_test.go

```diff
@@ -455,7 +454,9 @@ func testCompactWithStoreGateway(t *testing.T, penaltyDedup bool) {
 		Type:   client.S3,
 		Config: e2ethanos.NewS3Config(bucket, m.InternalEndpoint("https"), e2ethanos.ContainerSharedDir),
 	}
-	str, err := e2ethanos.NewStoreGW(e, "1", svcConfig, "")
+
+	// Crank down the deletion mark delay since deduplication can miss blocks in the presence of replica labels it doesn't know about.
+	str, err := e2ethanos.NewStoreGW(e, "1", svcConfig, "", []string{"--ignore-deletion-marks-delay=2s"})
```

Comment:
Do we now depend on this test to run faster than 2s?

Reply:
No, it now _cannot_ run faster than 2s.

Reply:
To be more clear, this is working around the bit mentioned in the description where store gateways load duplicates for a bit in the presence of replica labels. It cranks the window during which we ignore deletion marks down so that the test can pass once we recognize the deletion marks.